daemon/cmd/cells.go

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/gops"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/k8s"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/node"
+)
+
+var (
+	Agent = cell.Module(
+		"Cilium Agent",
+
+		Infrastructure,
+		ControlPlane,
+		Datapath,
+	)
+
+	// Infrastructure provides access and services to the outside.
+	// A cell should live here instead of ControlPlane if it is not needed by
+	// integrations tests, or needs to be mocked.
+	Infrastructure = cell.Module(
+		"Infrastructure",
+
+		// Runs the gops agent, a tool to diagnose Go processes.
+		gops.Cell(defaults.GopsPortAgent),
+
+		// Provides Clientset, API for accessing Kubernetes objects.
+		k8sClient.Cell,
+	)
+
+	// ControlPlane implement the per-node control functions. These are pure
+	// business logic and depend on datapath or infrastructure to perform
+	// actions. This separation enables non-privileged integration testing of
+	// the control-plane.
+	ControlPlane = cell.Module(
+		"Control Plane",
+
+		// LocalNodeStore holds onto the information about the local node and allows
+		// observing changes to it.
+		node.LocalNodeStoreCell,
+
+		// Shared resources provide access to k8s resources as event streams or as
+		// read-only stores.
+		k8s.SharedResourcesCell,
```

Comment:
Not sure if this should be considered "Infrastructure" or not. It depends on k8s client and our current tests mock that one using fake clients. We could also consider instead mocking `Resource[T]`, but not yet sure if it buys us much. So for now, it's easier to leave this in `ControlPlane`.